File: generators/manticoreNames.py
__all__ = ['manticoreNames']

# Don't look below, you will not understand this Python code :) I don't.

from js2py.pyjs import *
# setting scope
var = Scope( JS_BUILTINS )
set_global_object(var)

# Code follows:
var.registers(['nm1', 'nm4', 'nameGen', 'nm5', 'nm3', 'nm2'])
@Js
def PyJsHoisted_nameGen_(type, this, arguments, var=var):
    var = Scope({'arguments':arguments, 'this':this, 'type':type}, var)
    var.registers(['result', 'type'])
    var.put('result', Js([]))
    #for JS loop
    var.put('i', Js(0.0))
    while (var.get('i')<Js(10.0)):
        try:
            var.put('rnd', ((var.get('Math').callprop('random')*var.get('nm1').get('length'))|Js(0.0)))
            var.put('rnd2', ((var.get('Math').callprop('random')*var.get('nm2').get('length'))|Js(0.0)))
            var.put('rnd3', ((var.get('Math').callprop('random')*var.get('nm5').get('length'))|Js(0.0)))
            while PyJsStrictEq(var.get('nm1').get(var.get('rnd')),var.get('nm5').get(var.get('rnd3'))):
                var.put('rnd3', ((var.get('Math').callprop('random')*var.get('nm5').get('length'))|Js(0.0)))
            if (var.get('i')<Js(3.0)):
                while (var.get('rnd')<Js(5.0)):
                    var.put('rnd', ((var.get('Math').callprop('random')*var.get('nm1').get('length'))|Js(0.0)))
                while (var.get('rnd3')<Js(6.0)):
                    var.put('rnd3', ((var.get('Math').callprop('random')*var.get('nm5').get('length'))|Js(0.0)))
                var.put('names', ((var.get('nm1').get(var.get('rnd'))+var.get('nm2').get(var.get('rnd2')))+var.get('nm5').get(var.get('rnd3'))))
            else:
                var.put('rnd4', ((var.get('Math').callprop('random')*var.get('nm3').get('length'))|Js(0.0)))
                var.put('rnd5', ((var.get('Math').callprop('random')*var.get('nm2').get('length'))|Js(0.0)))
                while PyJsStrictEq(var.get('nm3').get(var.get('rnd4')),var.get('nm5').get(var.get('rnd3'))):
                    var.put('rnd4', ((var.get('Math').callprop('random')*var.get('nm3').get('length'))|Js(0.0)))
                if (var.get('i')<Js(7.0)):
                    var.put('names', ((((var.get('nm1').get(var.get('rnd'))+var.get('nm2').get(var.get('rnd2')))+var.get('nm3').get(var.get('rnd4')))+var.get('nm2').get(var.get('rnd5')))+var.get('nm5').get(var.get('rnd3'))))
                else:
                    var.put('rnd6', ((var.get('Math').callprop('random')*var.get('nm4').get('length'))|Js(0.0)))
                    var.put('rnd7', ((var.get('Math').callprop('random')*var.get('nm2').get('length'))|Js(0.0)))
                    while PyJsStrictEq(var.get('nm4').get(var.get('rnd6')),var.get('nm5').get(var.get('rnd3'))):
                        var.put('rnd6', ((var.get('Math').callprop('random')*var.get('nm5').get('length'))|Js(0.0)))
                    var.put('names', ((((((var.get('nm1').get(var.get('rnd'))+var.get('nm2').get(var.get('rnd2')))+var.get('nm3').get(var.get('rnd4')))+var.get('nm2').get(var.get('rnd5')))+var.get('nm4').get(var.get('rnd6')))+var.get('nm2').get(var.get('rnd7')))+var.get('nm5').get(var.get('rnd3'))))
            var.get('result').callprop('push', var.get('names'))
        finally:
                (var.put('i',Js(var.get('i').to_number())+Js(1))-Js(1))
    pass
    return var.get('result')
PyJsHoisted_nameGen_.func_name = 'nameGen'
var.put('nameGen', PyJsHoisted_nameGen_)
var.put('nm1', Js([Js(''), Js(''), Js(''), Js(''), Js(''), Js('b'), Js('ch'), Js('d'), Js('g'), Js('gh'), Js('h'), Js('j'), Js('k'), Js('kh'), Js('m'), Js('n'), Js('r'), Js('sh'), Js('t'), Js('v'), Js('y'), Js('z')]))
var.put('nm2', Js([Js('a'), Js('a'), Js('a'), Js('e'), Js('i'), Js('u'), Js('u'), Js('a'), Js('a'), Js('a'), Js('e'), Js('i'), Js('u'), Js('u'), Js('a'), Js('a'), Js('a'), Js('e'), Js('i'), Js('u'), Js('u'), Js('a'), Js('a'), Js('a'), Js('e'), Js('i'), Js('u'), Js('u'), Js('ia'), Js('ou'), Js('oo'), Js('ee')]))
var.put('nm3', Js([Js('fr'), Js('hk'), Js('hm'), Js('hr'), Js('kb'), Js('khg'), Js('kr'), Js('lt'), Js('lv'), Js('m'), Js('n'), Js('pr'), Js('r'), Js('rh'), Js('rkh'), Js('rm'), Js('rr'), Js('rsh'), Js('rt'), Js('ry'), Js('rz'), Js('sh'), Js('shk'), Js('t'), Js('v'), Js('vg'), Js('z'), Js('zy')]))
var.put('nm4', Js([Js('b'), Js('dd'), Js('dv'), Js('dy'), Js('gm'), Js('gn'), Js('j'), Js('khz'), Js('lf'), Js('ll'), Js('md'), Js('nb'), Js('nd'), Js('ndy'), Js('nsh'), Js('r'), Js('rb'), Js('rd'), Js('rt'), Js('rz'), Js('s'), Js('shy'), Js('v'), Js('y'), Js('zh'), Js('zm')]))
var.put('nm5', Js([Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js('b'), Js('d'), Js('hr'), Js('k'), Js('kh'), Js('n'), Js('nd'), Js('r'), Js('rd'), Js('rz'), Js('sh'), Js('shk'), Js('v'), Js('z'), Js('zl')]))
pass
pass


# Add lib to the module scope
manticoreNames = var.to_python()

File: js/manticoreNames.js
var nm1 = ["","","","","","b","ch","d","g","gh","h","j","k","kh","m","n","r","sh","t","v","y","z"];
var nm2 = ["a","a","a","e","i","u","u","a","a","a","e","i","u","u","a","a","a","e","i","u","u","a","a","a","e","i","u","u","ia","ou","oo","ee"];
var nm3 = ["fr","hk","hm","hr","kb","khg","kr","lt","lv","m","n","pr","r","rh","rkh","rm","rr","rsh","rt","ry","rz","sh","shk","t","v","vg","z","zy"];
var nm4 = ["b","dd","dv","dy","gm","gn","j","khz","lf","ll","md","nb","nd","ndy","nsh","r","rb","rd","rt","rz","s","shy","v","y","zh","zm"];
var nm5 = ["","","","","","","b","d","hr","k","kh","n","nd","r","rd","rz","sh","shk","v","z","zl"];

var br = "";

function nameGen(type){
	$('#placeholder').css('textTransform', 'capitalize');
	var element = document.createElement("div");
	element.setAttribute("id", "result");
	
	for(i = 0; i < 10; i++){
		rnd = Math.random() * nm1.length | 0;
		rnd2 = Math.random() * nm2.length | 0;
		rnd3 = Math.random() * nm5.length | 0;
		while(nm1[rnd] === nm5[rnd3]){
			rnd3 = Math.random() * nm5.length | 0;
		}
		if(i < 3){
			while(rnd < 5){
				rnd = Math.random() * nm1.length | 0;
			}
			while(rnd3 < 6){
				rnd3 = Math.random() * nm5.length | 0;
			}
			names = nm1[rnd] + nm2[rnd2] + nm5[rnd3];
		}else{
			rnd4 = Math.random() * nm3.length | 0;
			rnd5 = Math.random() * nm2.length | 0;
			while(nm3[rnd4] === nm5[rnd3]){
				rnd4 = Math.random() * nm3.length | 0;
			}
			if(i < 7){
				names = nm1[rnd] + nm2[rnd2] + nm3[rnd4] + nm2[rnd5] + nm5[rnd3];
			}else{
				rnd6 = Math.random() * nm4.length | 0;
				rnd7 = Math.random() * nm2.length | 0;
				while(nm4[rnd6] === nm5[rnd3]){
					rnd6 = Math.random() * nm5.length | 0;
				}
				names = nm1[rnd] + nm2[rnd2] + nm3[rnd4] + nm2[rnd5] + nm4[rnd6] + nm2[rnd7] + nm5[rnd3];
			}
		}
		br = document.createElement('br');	
		element.appendChild(document.createTextNode(names));
		element.appendChild(br);
	}
	if(document.getElementById("result")){
		document.getElementById("placeholder").removeChild(document.getElementById("result"));
	}		
	document.getElementById("placeholder").appendChild(element);
}